### ISSUE TITLE
Consider defaulting to "Mednafen Mode" in Ocotoshock

### ISSUE THREAD

**Issue by Zinfidel** [TASVideos/BizHawk#2899]
The default distribution of BizHawk starts Ocotoshock in the "Pixel Pro" mode which, which basically just doubles the framebuffer size and does no pixel aspect ratio correction (square pixels). Mednafen mode is the more console/tv-accurate setting that simulates the PAR.

On behalf of spikestuff
![image](https://user-images.githubusercontent.com/1929934/128555627-82a1f461-c056-485a-ba2a-944d7d31aea0.png)

**Comment by nattthebear** [comment]
I agree that the defaults should be geared more towards end users and less towards power users.

**Comment by zeromus** [comment]
Mednafen mode was designed to work with bilinear filtering, which we will NOT be enabling by default. If you guys actually like the non-filtered (tweaked) mednafen mode and want to enable it because some streamer somewhere articulated ignorance of the existence of options, that's up to you. I do not generally agree that we should be geared more towards end users (we made this decision on day one when we put the name BIZ in the project) but I don't generally disagree either. So except for the wrinkle I mentioned, I have no opinion.

**Comment by Zinfidel** [comment]
> Mednafen mode was designed to work with bilinear filtering

I did not know that this was the case. I agree that if this change is made that the tweaked mode should be preferred.

My opinion on this change is also not very strong - I know that some other cores in BizHawk are set to account for non-square pixels by default (Intellivision comes to mind) but I don't know how pervasive this predilection actually is among the cores. If the default for other cores leans towards TV-accuracy, then I think it makes sense here as well.

**Comment by Spikestuff** [comment]
> On behalf of spikestuff

Also Spikestuff: _Note: I'm using the word "issue" lightly, since this is stuff the user should be doing, but then again this should be something that's default (like controller)._
(pointing this missing line out since zeromus makes a comment that I am quoting in a second which is in agreeance)

Just to note this line.
> and want to enable it because some streamer somewhere articulated ignorance of the existence of options, that's up to you.

There's a reason why I also mentioned the controller patchup as well cause we had submissions where people were not bothering with checking their controller settings and having both controllers in, and an extra memory card in.
Making the file larger than it should be, and actually having to be checked in the bios sequence making it slower for a TAS by frames.
(I'm basically agreeing with user ignorance and explaining further of another case we had which was basically corrected to avoid).

But regardless, yes I will agree that I guess _Tweaked_ would be the preferred default option in the resolution management.

**Comment by YoshiRulz** [comment]
Defaulting to 1 connected gamepad is uncontroversial because users will quickly notice when they're unable to bind P2's controls. Would you put this resolution mode setting in the same group? Based on the comments in this thread I would.

**Comment by Meerkov** [comment]
As a dumb person, I would prefer if the default mode was Mednafen. PixelPro looks really weird when recording footage, and often can't record at 60 FPS because "it's slow".

I don't know who needs Pixel Pro mode, but I haven't yet run into a situation where I felt I needed it.

**Comment by Mad-Matty** [comment]
I see bizhawk as a tool. Specifically for Tasing but also for gameplay recording.  Retroarch exists if you want a more casual emu.  Pixelpro is good for accuracy and showing a pixel perfect unscaled image before any aspect correction is done.  Mednafen scaling might make a nicer tv like image but its not accurate and the original pixels are scaled, warped and not fit for archiving.

**Comment by EZGames69** [comment]
> Mednafen scaling might make a nicer tv like image but its not accurate and the original pixels are scaled, warped and not fit for archiving.

I’m confused by this statement.  If it’s more accurate to show internal resolution for the console, but it’s not accurate to show the image stretched to 4:3 like it would if you played it on a TV, then are you saying the intended way to view the PSX game is not on a TV?

**Comment by zeromus** [comment]
You're confused because you think there's a universal definition of "accurate".

**Comment by EZGames69** [comment]
My definition of “accurate” is how the game is supposed to be displayed on a TV

**Comment by zeromus** [comment]
Great, thanks for sharing. Other people have other definitions, even multiple for varying purposes.

**Comment by Meerkov** [comment]
Rather than argue about what the right definition of accurate is, I think I'll just say that we should follow the "principal of least surprise".

I think, unless you happen to be an emulation expert, then PixelPro mode is a "surprise". It's not how a user would normally have experienced the game. I've even filed bugs in this very repo complaining about how recording on PSX is slow, only to be told that it was because the default is PixelPro.

Despite my earlier self-deprecating statement, I'm not exactly a casual user. I TAS, reverse engineer, mod, record, stream... I've submitted patches to BizHawk multiple times... And this default setting was "surprising" to me. I mean that literally in that I didn't know that it existed, or that I would need to modify the default to get the behavior that I expected until *after* I submitted a bug to this repro and it was pointed out to me. If that surprises me, then I'm sure it surprises 90%+ of users. A comparable setting doesn't exist for SNES etc. Unless you're digging through menus, it's unlikely that you'll even realize it's there until *after* you've realized the default looks squished for some reason.

**Comment by zeromus** [comment]
I'm not convinced. 

1. Users don't expect their pixels to be nonuniformly munged in order to create the proper aspect ratio, either. 
2. Most emulators over history have worked as pixel pro mode does (or worse, without its internal doubling)
3. A comparable setting DOES exist for SNES (in the form of 'always double-size framebuffer on bsnes or 'constant framebuffer size' along with 'correct aspect ratio' in faust) and 'pad screen to 320' on GPGX core along with config > display > maintain aspect ratio and/or use 1:1 pixel size
4. Users are not likely to have experienced perfectly aspect-correct games on televisions. If you want to talk about what people experienced on their television, let's talk about how it was a blurry, jittery mess. If people expect that, they can expect to be disappointed as we have little interest in replicating that experience.

You're trying to do something complicated. Don't try to impress me by telling me it's too complicated. I'm likely to take it as praise for my great success, especially when you evidently havent even begun yet to appreciate the depth of its complication. We didn't set out to make something simple.

Here's my principle of least surprise: pixels surprisingly munged the least.

**Comment by Meerkov** [comment]
Mmm, I think maybe you're missing the idea of the Principle of Least Surprise. It's not that I don't appreciate that there are some reasons a pro user would want PixelPro mode, I just don't understand why it's the default. I'm certainly happy the option exists.

**Comment by zeromus** [comment]
OK, I'll make it simpler for you. You wanted to use the Principle of Least Surprise instead of Principle of Muh Accuracy, because you mistakenly believe it's not just as subject to arguments over what's surprising to users and who users are as Muh Accuracy is to arguments over what's accurate.

**Comment by TizianoMattei** [comment]
I find tweaked mednafen mode to not leave any space for lua scripting display on screen, which is the sole purpose i use bizhawk when i need to research / tas, not as duck/retroarch to just enjoy games.

**Comment by zeromus** [comment]
`client.SetGameExtraPadding(200, 0, 200, 0)`
this thread is going downhill fast

**Comment by TizianoMattei** [comment]
Another unnecessary workaround, forcing to rework multiple scripts, but at that point one would just switch to pixel pro each release first config and add another step. I guess i just dont understand why biz would be made more oriented towards enjoyable casual gameplay over research due to its current state in the first place, but thats a biased point of view, for intellectual honesty

> this thread is going downhill fast

So just like psx desyncs
